pandas/io/excel/_base.py

```diff
@@ -342,37 +350,136 @@
 )
 
 
+@overload
```

Comment:
Maybe add  a small comment to each overload to very briefly describe/highlight the differences between them. Could be as simple as `# sheet_name: str | int -> DataFrame`

Reply:
or add an inline comment to mark lines that change.

Reply:
Added an inline comment